src/SprykerEco/Zed/Braintree/Persistence/BraintreeEntityManager.php

```diff
@@ -89,6 +90,24 @@ public function updateIsShipmentOperationValue(int $idPaymentBraintree, string $
         }
     }
 
+    /**
+     * @param int $idSalesOrder
+     * @param \Generated\Shared\Transfer\BraintreePaymentTransfer $braintreePaymentTransfer
+     *
+     * @return void
+     */
+    public function updatePaymentByIdSalesOrder(int $idSalesOrder, BraintreePaymentTransfer $braintreePaymentTransfer): void
```

Comment:
Because 
`$idSalesOrder = $saveOrderTransfer->getIdSalesOrderOrFail();`
`$braintreePaymentTransfer = $paymentTransfer->getBraintreeOrFail();`
we take these from different transfers.